src/collection-facets/more-facets-content.ts

```diff
@@ -0,0 +1,365 @@
+  @property({ type: Object }) searchService?: SearchServiceInterface;
+
+  @property({ type: Object })
+  collectionNameCache?: CollectionNameCacheInterface;
+
+  @property({ type: Object })
+  languageCodeHandler?: LanguageCodeHandlerInterface;
+
+  @property({ type: Object }) selectedFacets?: SelectedFacets;
+
+  @state() aggregations?: Record<string, Aggregation>;
+
+  @state() castedBuckets?: Bucket[] = [];
+
+  @state() pageNumber = 1;
+
+  /**
+   * Facets are loading on popup
+   */
+  @state() facetsLoading = true;
+
+  @state() paginationSize = 1;
+
+  private facetsPerPage = 60; // Q. how many items we want to have on popup view
+
+  async updated(changed: PropertyValues) {
+    if (changed.has('facetKey')) {
+      this.facetsLoading = true;
+      this.pageNumber = 1;
+
+      await this.fetchSpecificFacets(
+        this.facetAggregationKey as unknown as string
+      );
+      this.facetsLoading = false;
+    }
+
+    if (changed.has('aggregations')) {
+      this.filterFacets();
+    }
+  }
+
+  async fetchSpecificFacets(facetAggregationKey: string) {
+    const aggregations = {
+      advancedParams: [
+        {
+          field: facetAggregationKey,
+          size: 1000000, // todo - do we want to have all the records at once?
+        },
+      ],
+    };
+
+    const params: SearchParams = {
+      query: this.fullQuery as string,
+      fields: ['identifier'],
+      aggregations,
+      rows: 1, // todo - do we want server-side pagination with offset/page/limit flag?
+    };
+
+    const results = await this.searchService?.search(params);
+    this.aggregations = results?.success?.response.aggregations as any;
+  }
+
+  async filterFacets() {
+    Object.entries(this.aggregations ?? []).forEach(([key, buckets]) => {
+      if (key === 'year_histogram') return;
+
+      this.castedBuckets = buckets['buckets'] as Bucket[];
+
+      if (this.facetKey === 'collection') {
+        // for collections, we need to asynchronously load the collection name
+        // so we use the `async-collection-name` widget and for the rest, we have a static value to use
+        const collectionIds = this.castedBuckets?.map(option => option.key);
+        const collectionIdsArray = Array.from(
+          new Set(collectionIds)
+        ) as string[];
+        this.collectionNameCache?.preloadIdentifiers(collectionIdsArray);
+      }
+    });
+
+    const { length } = Object.keys(this.castedBuckets as []);
+    this.paginationSize = Math.ceil(length / this.facetsPerPage);
+  }
+
+  private pageNumberClicked(e: CustomEvent<{ page: string }>) {
+    const page = e?.detail?.page;
+    if (page) {
+      this.pageNumber = Number(page);
+    }
+  }
+
+  private get renderMoreFacets() {
+    this.facetsLoading = false;
+    const min = (this.pageNumber - 1) * this.facetsPerPage;
+    const max = min + this.facetsPerPage - 1;
+
+    return html`<ul class="facet-list">
+      ${this.castedBuckets?.map((option, n) => {
+        const optionWrapperClass =
```

Comment:
idk what this all means?

Reply:
Thank you for putting my attention here. Currently it render all the items like 1800 items on DOM having visible/hidden class and switching that class with current page..

I did an improvement here to render only number of items per page. eg. currenlty we are showing 60 facets on one page.. so the map() will execute only 60 times on each page number click.